Txr/EncVelManager.cpp

```diff
@@ -51,17 +51,15 @@ void EncVelManager::SetLEDs(char force)
     ENC_GREEN_LED_OFF();
     ENC_RED_LED_OFF();
   }
-  // if we've crossed the midway point make sure lights are all the way off
-  else if (mPrevVelocityPercent < MID_VELOCITY ^ mVelocityPercent < MID_VELOCITY) {
-    SetAllLEDsOff();
-  }
   
   if (mVelocityPercent == MID_VELOCITY) {
+    SetAllLEDsOff();
     ENC_GREEN_LED_ON();
     ENC_RED_LED_ON();
   }
   else if (mVelocityPercent > MID_VELOCITY) {
     int dim = map(mVelocityPercent, MID_VELOCITY + 1, MAX_VELOCITY, 1, 250);
+    analogWrite(SPEED_LED2, 0);
```

Comment:
Is this necessary if you leave the original check in on line 55? With the fix above the lights get turned off and I can't get them to stick on any longer, nor can I think of how they would.  I'm pretty sure the reason they were sticking on before is when, in your fast move, you happen to get sampled right at MID_VELOCITY, which the XOR doesn't handle well.

My concern with adding this extra layer of insurance that the light will get turned off is that it's more CPU cycles most every time through this loop.

What do you think?